python/pyspark/worker.py

```diff
@@ -159,27 +226,13 @@ def wrapped(left_key_series, left_value_series, right_key_series, right_value_se
             key_series = left_key_series if not left_df.empty else right_key_series
             key = tuple(s[0] for s in key_series)
             result = f(key, left_df, right_df)
-        if not isinstance(result, pd.DataFrame):
-            raise TypeError(
-                "Return type of the user-defined function should be "
-                "pandas.DataFrame, but is {}".format(type(result))
-            )
-        # the number of columns of result have to match the return type
-        # but it is fine for result to have no columns at all if it is empty
-        if not (
-            len(result.columns) == len(return_type) or len(result.columns) == 0 and result.empty
-        ):
-            raise RuntimeError(
-                "Number of columns of the returned pandas.DataFrame "
-                "doesn't match specified schema. "
-                "Expected: {} Actual: {}".format(len(return_type), len(result.columns))
-            )
+        verify_pandas_result(result, return_type, assign_cols_by_name(runner_conf))
```

Comment:
I guess we can call `assign_cols_by_name(runner_conf)` outside of `wrapped`?

Reply:
Yes we could, but what is the benefit of moving that out of `wrapped`?

Reply:
`wrapped` will be called many times. We want to reduce as much overhead as possible.

Reply:
Very good point, thanks for clarification.

Reply:
done